### PR TITLE
fix: pull queries are supported on source streams now.

### PR DESCRIPTION
**Pull request by vvcephei**
### Description 
Correct the docs and add a test to prove it.

### Testing done 
New functional test.

### Reviewer checklist
- [ ] Ensure docs are updated if necessary. (eg. if a user visible feature is being added or changed).
- [ ] Ensure relevant issues are linked (description should include text like "Fixes #<issue number>")

### CODE REVIEW

**Review by JimGalasyn — APPROVED** [review]
LGTM!

**Review by cprasad1 — APPROVED** [review]

File: ksqldb-functional-tests/src/test/resources/rest-query-validation-tests/pull-queries-against-streams.json [hunk 1]
```diff
@@ -97,6 +97,35 @@
...
+      "name": "simple-source",
+      "properties": {
+        "ksql.query.pull.stream.enabled": true
+      },
+      "statements": [
+        "CREATE STREAM S1 (MYKEY INT KEY, MYVALUE INT) WITH (kafka_topic='test_topic', value_format='JSON');",
+        "SELECT * FROM S1;"
+      ],
+      "topics": [
+        {"name": "test_topic", "partitions": 1} // to get a stable ordering
```
[the remark this answers is not in the document]
Comment (vvcephei) [reply ksqldb-functional-tests/src/test/resources/rest-query-validation-tests/pull-queries-against-streams.json:110]: Yeah, I happened to notice it while I was trying to understand the rqtt schema a while back: https://github.com/confluentinc/ksql/blob/master/ksqldb-functional-tests/src/main/java/io/confluent/ksql/test/tools/TestJsonMapper.java#L44

File: ksqldb-functional-tests/src/test/resources/rest-query-validation-tests/pull-queries-against-streams.json [hunk 2]
```diff
@@ -97,6 +97,35 @@
         ]}
       ]
     },
+    {
+      "name": "simple-source",
+      "properties": {
+        "ksql.query.pull.stream.enabled": true
+      },
+      "statements": [
+        "CREATE STREAM S1 (MYKEY INT KEY, MYVALUE INT) WITH (kafka_topic='test_topic', value_format='JSON');",
```
[the remark this answers is not in the document]
Comment (vvcephei) [reply ksqldb-functional-tests/src/test/resources/rest-query-validation-tests/pull-queries-against-streams.json:106]: oops. I changed the name of the test and not the statement! That's what I get for rushing. I'll push the test fix change in a sec.

**Comment by vvcephei** [comment]
Pushed the test fix that Almog pointed out. I also noticed we were setting a property that's already the default.

File: ksqldb-functional-tests/src/test/resources/rest-query-validation-tests/pull-queries-against-streams.json [hunk 3]
```diff
@@ -5,9 +5,6 @@
   "tests": [
     {
       "name": "empty response on empty stream",
-      "properties": {
-        "ksql.query.pull.stream.enabled": true
```
Comment (mjsax) [inline ksqldb-functional-tests/src/test/resources/rest-query-validation-tests/pull-queries-against-streams.json:9]: Those are enabled by default?

**Review by mjsax — APPROVED** [review]

**Pull request merged**